httpx_auth/_oauth2/common.py

```diff
@@ -86,3 +87,32 @@ def request_new_grant_with_post(
 class OAuth2:
     token_cache = TokenMemoryCache()
     display = DisplaySettings()
+
+
+class OAuth2BaseAuth(abc.ABC, httpx.Auth):
+    state: Optional[str] = None
+    early_expiry: float
+
+    refresh_token: Optional[Callable]
```

Comment:
Nothing should be set per class, use the instance to avoid 2 properties of the same name coexisting with different values

Reply:
Yeah, the proposal is to switch it to parameters at instance init. Type hinting can be added to those parameters. It will avoid mistakes and make it clear how everything is set up instead of navigating in between classes